FrontEnd/src/components/SearchListComponent.js

```diff
@@ -0,0 +1,275 @@
+        return (
+            <Table>
+                <TableHead>
+                    <TableRow>
+                        {props.tableHeadWords.map((word) => {
+                            return(
+                                <TableCell align={"center"}>
+                                    <Typography variant={"h6"}>
+                                        {word}
+                                    </Typography>
+                                </TableCell>
+                            )
+                        })}
+                    </TableRow>
+                </TableHead>
+
+               {arrayOfDataRecords.map((dataRecord) => {
+                /*Der Nutzer wird nur an die TableRow gegeben, wenn entweder nicht gefiltert wird (Für die erste Liste)
+                ODER wenn die Eingabe im Suchfeld dem Parameter des Nutzers entspricht, der im Radiocontrol eingestellt ist
+                (z.B. username des Nutzers entspricht dem im Suchfeld) und zusätzlich die Rolle der im Rollenfilter eingestellten Rolle entspricht */
+                   if(!applyFilter||((dataRecord[selectedSearchOption].toLowerCase().includes(searchFieldValue.toLowerCase())) && (dataRecord[props.filterParameter]===selectedFilterOption||selectedFilterOption==="all"))) {
+                        return (
+                            <TableRow className={dataRecord.role==="teacher"?classesCustom.dataRowTeacher:classesCustom.dataRow}>
+                                {/*Es wird hier über alle Einträge des dataRecord iteriert, uns aus jedem Eintrag wird eine Tabellenzelle gebastelt*/}
+                                {Object.values(dataRecord).map((dataRecordField) => {
+                                    return (
+                                        <TableCell align={"center"}>
+                                            <Typography >
+                                                {dataRecordField}
+                                            </Typography>
+                                        </TableCell>
+                                    )
+                                })}
+
+                                {/*Es wird hier geprüft, ob für jeden Dateneintrag in der Liste ein Button zum verschieben in
+                                die jeweils andere Liste eingefügt werden soll. Dafür muss zuerst geprüft werden, ob es überhaupt
+                                eine TwoListComponent ist und zusätzlich, ob es denn die erste oder zweite Liste ist, um dann
+                                den entsprechenden Button einzufügen.
+                                Wenn nicht, sollen die Buttons eingefügt werden, die von den Props der Komponente zur
+                                Verfügung gestellt werden.
+                                Ja, verschachtelte tertiäre Ausdrücke innerhalb von JSX sucken... aber muss ja manchmal sein*/}
+                                {isFirstList&&props.isTwoListComponent?
```

Comment:
Könntest du auch in eine Fkt auslagern. const renderButtons = (die benötigten Params) => {
  hier der Code
}
und dann an dieser Stelle im Code nur noch {renderButtons()} einfügen